src/element/writer.rs

```diff
@@ -13,13 +13,12 @@ use crate::{Element, Value};
 
 /// Serializes [`Element`] instances into some kind of output sink.
 pub trait ElementWriter {
-    /// Serializes a single [`Element`] at the current depth of the writer.
-    fn write_element(&mut self, element: &Element) -> IonResult<()>;
-
     /// Serializes a single [`Value`] at the current depth of the writer.
-    // TODO: consider extracting this to a ValueWriter trait.
     fn write_value(&mut self, value: &Value) -> IonResult<()>;
 
+    /// Serializes a single [`Element`] at the current depth of the writer.
+    fn write_element(&mut self, element: &Element) -> IonResult<()>;
+
```

Comment:
Is this change intentionally part of this PR?

Reply:
Oops! The change was intentional, but not meant for this PR. Good eye.